### PR TITLE
External Remove Proposal

### PR DESCRIPTION
**Pull request by augustocdias**

### CODE REVIEW

File: openmls/src/group/tests/external_remove_proposal.rs [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+            group_id,
+            credential_with_keys.credential_with_key.clone(),
+        )
+        .unwrap(),
+        credential_with_keys,
+    )
+}
+
+// Validation test setup
+fn validation_test_setup(
```
Comment (beltram) [inline openmls/src/group/tests/external_remove_proposal.rs:49]: You (almost) never use this helper although you always need in your tests a group with Alice & Bob

File: openmls/src/group/tests/external_remove_proposal.rs [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+    let bob_index = alice_group
+        .members()
+        .find_map(|member| {
+            if member.credential.identity() == b"Bob" {
+                Some(member.index)
+            } else {
+                None
+            }
+        })
+        .unwrap();
```
Comment (beltram) [inline openmls/src/group/tests/external_remove_proposal.rs:170]:
```suggestion
let bob_index = alice_group
        .members()
        .find(|m| m.credential.identity() == b"Bob")
        .map(|m| m.index)
        .unwrap();
```

**Review by beltram — CHANGES_REQUESTED** [review]

File: openmls/src/extensions/mod.rs [hunk 3]
```diff
@@ -42,7 +42,7 @@ pub mod errors;
 // Public re-exports
 pub use application_id_extension::ApplicationIdExtension;
 pub use external_pub_extension::ExternalPubExtension;
-pub use external_sender_extension::ExternalSendersExtension;
+pub use external_sender_extension::{ExternalSender, ExternalSendersExtension};
```
[the remark this answers is not in the document]
Comment (augustocdias) [reply openmls/src/extensions/mod.rs:45]: this is a re-export. Without this it is impossible for consumers to actually add `ExternalSenders` into the the group's extensions.

File: openmls/src/group/core_group/process.rs [hunk 4]
```diff
@@ -214,10 +215,33 @@ impl CoreGroup {
                     Some(credential),
                 ))
             }
-            UnverifiedContextMessage::External(_external_message) => {
-                // We don't support messages from external senders yet
-                // TODO #151/#106
-                todo!()
+            UnverifiedContextMessage::External(external_message) => {
+                let verified_external_message = external_message.into_verified(backend.crypto())?;
```
[the remark this answers is not in the document]
Comment (augustocdias) [reply openmls/src/group/core_group/process.rs:219]: There's no error variant introduced here and failure's here will result in `InvalidSignature`. Did you mean to comment on this line?

**Pull request closed without merging**